workitem/workitem_repository.go

```diff
@@ -1131,3 +1161,196 @@ func (r *GormWorkItemRepository) LoadByIteration(ctx context.Context, iterationI
+				delete(wiStorage.Fields, oldFieldName)
+			}
+		} else { // field doesn't exist in new type
+			if wiStorage.Fields[oldFieldName] != nil {
+				fieldDiff[oldFieldName] = wiStorage.Fields[oldFieldName]
+				delete(wiStorage.Fields, oldFieldName)
+			}
+		}
+	}
+	// We need fieldKeys to show field diff in a defined order. Golang maps
+	// aren't ordered by default.
+	var fieldKeys []string
+	for fieldName := range fieldDiff {
+		fieldKeys = append(fieldKeys, fieldName)
+	}
+	// Sort the field keys to prevent random order of fields
+	sort.Strings(fieldKeys)
+	// Append diff (fields along with their values) between the workitem types
+	// to the description
+	if len(fieldDiff) > 0 {
+		originalDescription := rendering.NewMarkupContentFromValue(wiStorage.Fields[SystemDescription])
+		textToPrepend := "\n\n======= Missing fields in workitem type: " + newWIType.Name + " =======\n"
+		for _, fieldName := range fieldKeys {
+			fieldDef := oldWIType.Fields[fieldName]
+			oldKind := fieldDef.Type.GetKind()
+			oldValue := fieldDiff[fieldName]
+
+			if oldKind == KindEnum {
+				enumType, ok := fieldDef.Type.(EnumType)
+				if !ok {
+					return errs.Errorf("failed to convert field \"%s\" to enum type: %+v", fieldName, fieldDef)
+				}
+				oldKind = enumType.BaseType.GetKind()
+			}
+			// handle all single value fields (including Enums)
+			if oldKind != KindList {
+				var val string
+				if oldKind.IsRelational() {
+					val, err = getValueOfRelationalKind(r.db, oldValue, oldKind)
+					if err != nil {
+						return errs.Wrapf(err, "failed to get relational value for field %s", fieldName)
+					}
+				} else {
+					val = fmt.Sprint(oldValue)
+				}
+				textToPrepend += fmt.Sprintf("\n\n %s : %s", oldWIType.Fields[fieldName].Label, val)
+				continue
+			}
+
+			// Deal with multi value field (KindList)
+			listType, ok := fieldDef.Type.(ListType)
+			if !ok {
+				return errs.Errorf("failed to convert field \"%s\" to list type: %+v", fieldName, fieldDef)
+			}
+			oldKind = listType.ComponentType.GetKind()
+			valList, ok := fieldDiff[fieldName].([]interface{})
+			if !ok {
+				return errs.Errorf("failed to convert list value of field \"%s\" to []interface{}: %+v", fieldName, fieldDiff[fieldName])
+			}
+
+			var tempList []string
+			for _, v := range valList {
+				val := fmt.Sprint(v)
+				if oldKind.IsRelational() {
+					val, err = getValueOfRelationalKind(r.db, v, oldKind)
+					if err != nil {
+						return errs.Wrapf(err, "failed to get relational value for field %s", fieldName)
+					}
+				}
+				tempList = append(tempList, val)
+			}
+			// Convert []string to comma seperated strings.
+			textToPrepend += fmt.Sprintf("\n\n %s : %s", oldWIType.Fields[fieldName].Label, strings.Join(tempList, ", "))
+		}
+		textToPrepend += "\n\n================================================\n\n"
+		wiStorage.Fields[SystemDescription] = *(rendering.NewMarkupContentFromValue(textToPrepend + originalDescription.Content))
```

Comment:
There should be a check if workitem has description otherwise `originalDescription.Content` will result in runtime error

Reply:
Please see https://github.com/fabric8-services/fabric8-wit/pull/2202/commits/5ee62b25ac0b6851eb25ff39a605e2c205a875b9

I have assigned empty description to the workitem with no description. This would ensure we never get a panic

Reply:
@jarifibrahim Please add test to capture this scenario